### PR TITLE
[reddit] test of user and submission conversion to activity stream objects

### PR DESCRIPTION
**Pull request by stedn**
Here is a test of the conversion from praw objects to activitystream objects.  I have written fake classes that contain the attributes needed from the praw models.

I could see that coveralls had dropped by 1% on the last merge.  How can I check if this gets it back up?

### CODE REVIEW

**Comment by snarfed** [comment]
> I could see that coveralls had dropped by 1% on the last merge. How can I check if this gets it back up?

ooh thanks for looking! you'd click _Show all checks_ above, then _Details_ to get to the CircleCI build, then click on the _Artifacts_ tab, which has the raw HTML coverage output:

https://2109-3557653-gh.circle-artifacts.com/0/tmp/coverage_html/index.html

coveralls itself doesn't run on fork builds, so that its token isn't exposed to untrusted code, but once the PR is merged, you can find the coveralls build URL at the bottom of the _Python 3_ build step log in CircleCI.

File: granary/tests/test_reddit.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+from granary import reddit
+
+import copy
+import json
+
+class FakeRedditor():
+  id = '59ucsixw'
+  name = 'bonkerfield'
+  subreddit = {'display_name': 'u_bonkerfield',
```
Comment (snarfed) [inline granary/tests/test_reddit.py:12]: minor style nit, newline after `{` pls

File: granary/tests/test_reddit.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+   'title': '',
+   'icon_img': 'https://styles.redditmedia.com/t5_2az095/styles/profileIcon_ek6onop1xbf41.png?width=256&height=256&crop=256:256,smart&s=81fbf5a84c69533b5679f46638353d9e8471b817',
```
Comment (snarfed) [inline granary/tests/test_reddit.py:14]: minor, pls remove the `s` query param? may be innocuous but looks like a token. same elsewhere

File: granary/tests/test_reddit.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
+from oauth_dropins.webutil import testutil
+from oauth_dropins.webutil import util
+
+from granary import reddit
+
+import copy
+import json
+
+class FakeRedditor():
```
Comment (snarfed) [inline granary/tests/test_reddit.py:9]: mind adding a praw API object link and/or explanation of which API object this is faking? same below.

File: granary/tests/test_reddit.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      'objectType': 'person',
+      'displayName': 'bonkerfield',
+      'image': {'url': 'https://styles.redditmedia.com/t5_2az095/styles/profileIcon_ek6onop1xbf41.png?width=256&height=256&crop=256:256,smart&s=81fbf5a84c69533b5679f46638353d9e8471b817'},
+      'id': 'tag:reddit.com:bonkerfield',
+      'numeric_id': '59ucsixw',
+      'published': '2019-12-21T17:40:11Z',
+      'url': 'https://reddit.com/user/bonkerfield/',
+      'urls': [
+        {
+        "value":"https://reddit.com/user/bonkerfield/"
```
Comment (snarfed) [inline granary/tests/test_reddit.py:52]: minor style nit, indent pls, or inline like `[{"value": ...}]`

File: granary/tests/test_reddit.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        }
+      ],
+      'username': 'bonkerfield',
+      'description': 'https://bonkerfield.org https://viewfoil.bonkerfield.org',
+      }
+
+    self.assert_equals(actor, self.reddit.user_to_actor(FakeRedditor()))
+
+  def test_submission_to_activity(self):
+    activity = {
```
Comment (snarfed) [inline granary/tests/test_reddit.py:68]: minor, for consistency with other granary tests, mind moving this up into a module-level constant?

File: granary/tests/test_reddit.py [hunk 6]
```diff
@@ -7,22 +7,25 @@
 import json
 
 class FakeRedditor():
+  # see https://praw.readthedocs.io/en/latest/code_overview/models/redditor.html
```
Comment (snarfed) [inline granary/tests/test_reddit.py:10]: very minor: docstring instead of comment pls

File: granary/tests/test_reddit.py [hunk 7]
```diff
@@ -34,45 +37,31 @@ def __init__(self, fake_redditor):
...
-      'numeric_id': '59ucsixw',
-      'published': '2019-12-21T17:40:11Z',
-      'url': 'https://reddit.com/user/bonkerfield/',
-      'urls': [
-        {
-        "value":"https://reddit.com/user/bonkerfield/"
-        },
-        {
-        "value":"https://bonkerfield.org"
+  ACTIVITY = {
```
Comment (snarfed) [inline granary/tests/test_reddit.py:40]: minor: these go at module level, not class level

Comment (stedn) [reply granary/tests/test_reddit.py]: oops, misunderstood, fixed.

File: granary/tests/test_reddit.py (hunk 6 above)
Comment (stedn) [reply granary/tests/test_reddit.py]: ah, right, fixed.

**Pull request merged**